### PR TITLE
Add instructions on using a Docker image for development

### LINKED ISSUE #328: Provide documentation on how to do development on one of our docker images
We should be using this as a fall back for machines we don't support.  Basically put the series of commands to download the image and build our code using the libraries included in the docker image.  We could also find a short useful youtube tutorial for getting up and running with docker.

### LINKED ISSUE #359: Add xsltproc to the docker images
We use xsltproc to transform the CTest output to JUnit.  This might allow more information to show up in our PRs.

I added a check that doesn't fail the docker builds that outputs the following:

```
[Checking to see if xsltproc exists...]
[exe: xsltproc --help]
[ERROR: [return code: 127] from command: xsltproc --help]
[WARNING: xsltproc does not exist skipping JUnit conversion]
```

### PR DESCRIPTION
**Pull request by joshessman-llnl**
Resolves #328

https://serac.readthedocs.io/en/docs-essman-using_docker/sphinx/dev_guide/docker_dev_env.html

### CODE REVIEW

**Comment by jamiebramwell** [comment]
The docs are available for browsing [here](https://serac.readthedocs.io/en/docs-essman-using_docker/).

File: src/docs/sphinx/docker_info.rst [hunk 1]
```diff
@@ -7,6 +7,37 @@
...
+.. image:: copy_pull_cmd.png
+   :scale: 50 %
+
+4. Next, run the copied command.  Our images are around 2.5 GB, so it may take a while for the image to be downloaded to your machine.
+   When the download completes, you will see something like the following:
+
+.. image:: pull_complete.png
+
+5. You can now run the image.  Run ``docker run -it -v /your/serac/repo:/home/serac/serac seracllnl/tpls:gcc-8_12-15-20_20h-36m bash``,
+   replacing the tag (the part following the ``:``) with the tag you used in the ``docker pull`` command and
```
Comment (white238) [inline src/docs/sphinx/docker_info.rst]: This instruction is confusing, mainly due to two `:` in the command.

File: src/docs/sphinx/docker_info.rst [hunk 2]
```diff
@@ -7,6 +7,37 @@
...
+   replacing ``/your/serac/repo`` with the path to the Serac repo you cloned in the first step.  This will open a terminal into the image.
+
+#. Follow the build instructions detailed in :ref:`building-serac`, using the host-config in ``host-configs/docker`` that
+   corresponds to the compiler you've selected.  These commands should be run using the terminal you opened in the previous step.
+
+#. You can now develop normally your host machine (e.g., via a graphical text editor).
```
Comment (white238) [inline src/docs/sphinx/docker_info.rst]: "normally **on** your host machine"?

Also somewhere this needs clarification that you are mapping the local directory to the directory in the docker container

**Comment by jamiebramwell** [comment]
When I tried to do this, I got a permissions error. Any ideas? 
Screenshot 2021-01-15 135033

**Comment by joshessman-llnl** [comment]
> 
> 
> When I tried to do this, I got a permissions error. Any ideas?
> Screenshot 2021-01-15 135033

Thanks for trying this out!  I think there may need to be a `-u serac` in the `docker run` command - I'll update the docs accordingly.

**Review by jamiebramwell — APPROVED** [review]
Works on my machine! We should figure out why the volume is mounted as axom instead of serac in the docker container for a future PR.

**Comment by jamiebramwell** [comment]
@white238 Do you have any outstanding issues?

**Comment by jamiebramwell** [comment]
@jandrej @salazardetroya , since you are interested in using these instructions, does it make sense? Thanks for the help!

**Comment by jamiebramwell** [comment]
The documentation is built and hosted here: https://serac.readthedocs.io/en/docs-essman-using_docker/sphinx/dev_guide/docker_dev_env.html

**Comment by salazardetroya** [comment]
Hi, I've got serac and compiled and running with the docker image! I would only stress downloading **the most recent** image in the docs to prevent lazy people like me to simply copy and paste the instructions given there. Or maybe add an image with the `latest` tag? That is what I usually see in other projects.

**Comment by jamiebramwell** [comment]
An excellent point @salazardetroya ! I'll update the docs and look into adding a `latest` tag. Thanks for trying it out!

**Comment by joshessman-llnl** [comment]
I went ahead and updated the docker images to add `latest` tags for each compiler version, and since I was updating the images, I added `xsltproc` as well: https://dev.azure.com/llnl-serac/serac/_build/results?buildId=2932&view=logs&j=2742912e-f547-59b4-f336-65c28301f30c&t=cd7bfce2-9b19-5c4f-5f3a-913e42776d4e&l=12418

Resolves #359

**Comment by jamiebramwell** [comment]
@white238 can you take a look at this? I think it's been vetted enough to merge at this point.

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
LGTM

**Pull request merged**